hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java

```diff
@@ -496,6 +496,11 @@ public boolean delete(Path f, boolean recursive) throws IOException {
     String key = pathToKey(f);
     boolean result;
 
+    if (f.isRoot()) {
```

Comment:
There is a similar check below inside `innerDelete`-
```
    try {
      DeleteIterator iterator = new DeleteIterator(f, recursive);

      if (f.isRoot()) {
        LOG.warn("Cannot delete root directory.");
        return false;
      }
```
Does that not work? Or will that be redundant post this? 
In any case this should be inside `    if (status.isDirectory()) {` block

Reply:
Ok，I will add the unit test